Change 28510 by stevep@stevep-kirk on 2006/07/08 17:51:43

	Subject: [PATCH] perl5db.pl: read full lines from remote socket
	From: Brendan O'Dea <bod@debian.org>
	Date: Sat, 8 Jul 2006 13:27:50 +1000
	Message-ID: <20060708032750.GA20454@londo.c47.org>

Affected files ...

... //depot/perl/lib/perl5db.pl#119 edit

Differences ...

==== //depot/perl/lib/perl5db.pl#119 (text) ====

@@ -6354,9 +6354,13 @@
         $OUT->write( join( '', @_ ) );
 
         # Receive anything there is to receive.
-        my $stuff;
-        $IN->recv( $stuff, 2048 );    # XXX "what's wrong with sysread?"
-                                      # XXX Don't know. You tell me.
+        $stuff;
+        my $stuff = '';
+        my $buf;
+        do {
+            $IN->recv( $buf = '', 2048 );   # XXX "what's wrong with sysread?"
+                                            # XXX Don't know. You tell me.
+        } while length $buf and ($stuff .= $buf) !~ /\n/;
 
         # What we got.
         $stuff;

